tests/actionExecutor.spec.ts

```diff
@@ -120,6 +120,78 @@ describe("Action executor tests: ", () => {
 
     done();
   });
+
+  it("should retry action with timeout", async done => {
+    nock("http://localhost:8000")
+      .get("/swaps/rfc003/")
+      .reply(200, swapsAcceptDeclineStub);
+    const scope = nock("http://localhost:8000")
+      .post("/swaps/rfc003/399e8ff5-9729-479e-aad8-49b03f8fc5d5/accept")
+      .replyWithError("Oh no, we ran out of bananas.")
+      .post("/swaps/rfc003/399e8ff5-9729-479e-aad8-49b03f8fc5d5/accept")
+      .reply(200, acceptedStub);
+
+    const getData = jest.fn();
+    getData.mockReturnValueOnce("0xcb777414c38e426c7038a7c4908751f5e864f7ad");
+
+    const datastore: FieldDataSource = {
+      getData
+    };
+    const actionExecutor = new ActionExecutor(
+      comitNode,
+      datastore,
+      new DummyLedgerExecutor()
+    );
+    const swap = swapsAcceptDeclineStub.entities[0] as Swap;
+    const acceptAction = swap.actions.find(
+      action => action.name === "accept"
+    ) as Action;
+
+    const actionResponse = await actionExecutor.execute(acceptAction, 1, 2);
+
+    expect(actionResponse).toStrictEqual(Result.ok(acceptedStub));
+
+    expect(scope.isDone()).toBeTruthy();
+
+    done();
+  });
+
+  it("should retry action and still fail if nothing changed (no endless loop)", async done => {
+    nock("http://localhost:8000")
+      .get("/swaps/rfc003/")
+      .reply(200, swapsAcceptDeclineStub);
+    const scope = nock("http://localhost:8000")
+      .post("/swaps/rfc003/399e8ff5-9729-479e-aad8-49b03f8fc5d5/accept")
+      .replyWithError("Oh no, we ran out of bananas.")
+      .post("/swaps/rfc003/399e8ff5-9729-479e-aad8-49b03f8fc5d5/accept")
+      .replyWithError("You're telling me still no bananas?")
+      .post("/swaps/rfc003/399e8ff5-9729-479e-aad8-49b03f8fc5d5/accept")
+      .replyWithError("🙈🙉🙊");
+
+    const getData = jest.fn();
+    const actionExecutor = new ActionExecutor(
+      comitNode,
+      {
+        getData
+      },
+      new DummyLedgerExecutor()
+    );
+    const swap = swapsAcceptDeclineStub.entities[0] as Swap;
+    const acceptAction = swap.actions.find(
+      action => action.name === "accept"
+    ) as Action;
+
+    const actionResponse = await actionExecutor.execute(acceptAction, 2, 2);
+
+    expect(actionResponse).toStrictEqual(
+      Result.err(new Error("Maximum number of retries reached"))
+    );
+
+    // check the mock was invoked indeed 3 times
```

Comment:
```suggestion
    // check the mock finished execution as expected
```
It does not really check the exact number of times here (but they are implied).